components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -100,208 +58,30 @@ void BraveReferralsService::Start() {
         base::Bind(&BraveReferralsService::OnReadPromoCodeComplete,
                    weak_factory_.GetWeakPtr()));
+  }
 
   initialized_ = true;
 }
 
 void BraveReferralsService::Stop() {
-  fetch_referral_headers_timer_.reset();
   initialized_ = false;
 }
 
-// static
-bool BraveReferralsService::GetMatchingReferralHeaders(
-    const base::ListValue& referral_headers_list,
-    const base::DictionaryValue** request_headers_dict,
-    const GURL& url) {
-  // If the domain for this request matches one of our target domains,
-  // set the associated custom headers.
-  for (const auto& headers_value : referral_headers_list) {
-    const base::Value* domains_list =
-        headers_value.FindKeyOfType("domains", base::Value::Type::LIST);
-    if (!domains_list) {
-      LOG(WARNING) << "Failed to retrieve 'domains' key from referral headers";
-      continue;
-    }
-    const base::Value* headers_dict =
-        headers_value.FindKeyOfType("headers", base::Value::Type::DICTIONARY);
-    if (!headers_dict) {
-      LOG(WARNING) << "Failed to retrieve 'headers' key from referral headers";
-      continue;
-    }
-    for (const auto& domain_value : domains_list->GetList()) {
-      URLPattern url_pattern(URLPattern::SCHEME_HTTPS |
-                             URLPattern::SCHEME_HTTP);
-      url_pattern.SetScheme("*");
-      url_pattern.SetHost(domain_value.GetString());
-      url_pattern.SetPath("/*");
-      url_pattern.SetMatchSubdomains(true);
-      if (!url_pattern.MatchesURL(url))
-        continue;
-      return headers_dict->GetAsDictionary(request_headers_dict);
-    }
-  }
-  return false;
-}
-
-void BraveReferralsService::OnFetchReferralHeadersTimerFired() {
-  FetchReferralHeaders();
-}
-
-void BraveReferralsService::OnReferralHeadersLoadComplete(
-    std::unique_ptr<std::string> response_body) {
-  int response_code = -1;
-  if (referral_headers_loader_->ResponseInfo() &&
-      referral_headers_loader_->ResponseInfo()->headers)
-    response_code =
-        referral_headers_loader_->ResponseInfo()->headers->response_code();
-  if (referral_headers_loader_->NetError() != net::OK || response_code < 200 ||
-      response_code > 299) {
-    const std::string safe_response_body =
-        response_body ? *response_body : std::string();
-    LOG(ERROR) << "Failed to fetch headers from referral server"
-               << ", error: " << referral_headers_loader_->NetError()
-               << ", response code: " << response_code
-               << ", payload: " << safe_response_body
-               << ", url: " << referral_headers_loader_->GetFinalURL().spec();
-    return;
-  }
-
-  base::Optional<base::Value> root =
-      base::JSONReader().ReadToValue(*response_body);
-  if (!root || !root->is_list()) {
-    LOG(ERROR) << "Failed to parse referral headers response";
-    return;
-  }
-  pref_service_->Set(kReferralHeaders, root.value());
-}
-
-void BraveReferralsService::OnReferralInitLoadComplete(
```

Comment:
OnReferralInitLoadComplete should remain as that contains the functionality to handle successful referral initialization.